### PR TITLE
 Added documentation about passing custom disallowed types to Hash#from_xml [ci skip]

### PR DESCRIPTION
**Pull request by prathamesh-sonpatki**

### CODE REVIEW

File: activesupport/lib/active_support/core_ext/hash/conversions.rb [hunk 1]
```diff
@@ -106,7 +106,25 @@ class << self
...
+    #         <foo type="integer">1</foo>
+    #         <bar type="string">"David"</bar>
+    #       </hash>
+    #   XML
+    #
+    #   hash = Hash.from_xml(xml, ['integer'])
+    #   # => ActiveSupport::XMLConverter::DisallowedType: Disallowed type attribute: "integer"
+    #
+    # Note that the default disallowed types - Symbol and YAML will be
+    # ignored in case custom disallowed types are passed by user.
```
Comment (zzak) [inline activesupport/lib/active_support/core_ext/hash/conversions.rb]: Not sure I really understand this statement

File: activesupport/lib/active_support/core_ext/hash/conversions.rb [hunk 2]
```diff
@@ -106,7 +106,25 @@ class << self
...
+    #
+    #   xml = <<-XML
+    #     <?xml version="1.0" encoding="UTF-8"?>
+    #       <hash>
+    #         <foo type="integer">1</foo>
+    #         <bar type="string">"David"</bar>
+    #       </hash>
+    #   XML
+    #
+    #   hash = Hash.from_xml(xml, ['integer'])
```
Comment (zzak) [inline activesupport/lib/active_support/core_ext/hash/conversions.rb]: If the receiver has `*disallowed_types` I think we only need `from_xml(xml, 'integer')` in this case?

Comment (prathamesh-sonpatki) [inline activesupport/lib/active_support/core_ext/hash/conversions.rb]: @zzak Thanks for the review. The receiver does not have `*disallowed_types` currently. Are you suggesting to change code?

File: activesupport/lib/active_support/core_ext/hash/conversions.rb (hunk 1 above)
Comment (prathamesh-sonpatki) [inline activesupport/lib/active_support/core_ext/hash/conversions.rb]:
```ruby
def initialize(xml, disallowed_types = nil)
   @xml = normalize_keys(XmlMini.parse(xml))
   @disallowed_types = disallowed_types || DISALLOWED_TYPES
end
```
If I pass `[integer]` as custom disallowed type then default DISALLOWED_TYPES i.e. Symbol and YAML will be ignored due to the `||` condition.

Comment (zzak) [inline activesupport/lib/active_support/core_ext/hash/conversions.rb]: Ohhhh, [got it](https://github.com/rails/rails/blob/3e36db4406beea32772b1db1e9a16cc1e8aea14c/activesupport/lib/active_support/core_ext/hash/conversions.rb#L131-L134). :coffee: 

Can we say this instead?

> Note that passing custom disallowed types will override the default types, which are Symbol and YAML.

File: activesupport/lib/active_support/core_ext/hash/conversions.rb (hunk 2 above)
Comment (zzak) [inline activesupport/lib/active_support/core_ext/hash/conversions.rb]: Probably not worth the code change

Comment (prathamesh-sonpatki) [inline activesupport/lib/active_support/core_ext/hash/conversions.rb]: :+1:

File: activesupport/lib/active_support/core_ext/hash/conversions.rb (hunk 1 above)
Comment (prathamesh-sonpatki) [inline activesupport/lib/active_support/core_ext/hash/conversions.rb]: @zzak I have updated the comment :smile:

**Comment by zzak** [comment]
@prathamesh-sonpatki Thanks for your help! <3

**Comment by prathamesh-sonpatki** [comment]
Thank you.

**Pull request merged**